apis/eks/v1alpha1/types.go

```diff
@@ -0,0 +1,383 @@
+	//    while attempting to launch instances.
+	//
+	//    * NodeCreationFailure: Your launched instances are unable to register
+	//    with your Amazon EKS cluster. Common causes of this failure are insufficient
+	//    worker node IAM role (https://docs.aws.amazon.com/eks/latest/userguide/worker_node_IAM_role.html)
+	//    permissions or lack of outbound internet access for the nodes.
+	//
+	//    * InstanceLimitExceeded: Your AWS account is unable to launch any more
+	//    instances of the specified instance type. You may be able to request an
+	//    Amazon EC2 instance limit increase to recover.
+	//
+	//    * InsufficientFreeAddresses: One or more of the subnets associated with
+	//    your managed node group does not have enough available IP addresses for
+	//    new nodes.
+	//
+	//    * AccessDenied: Amazon EKS or one or more of your managed nodes is unable
+	//    to communicate with your cluster API server.
+	//
+	//    * InternalFailure: These errors are usually caused by an Amazon EKS server-side
+	//    issue.
+	Code NodeGroupIssueCode `json:"code,omitempty"`
+
+	// The error message associated with the issue.
+	Message string `json:"message,omitempty"`
+
+	// The AWS resources that are afflicted by this issue.
+	ResourceIDs []string `json:"resourceIds,omitempty"`
+}
+
+// NodeGroupResources describe resources in a NodeGroup.
+type NodeGroupResources struct {
+	// The Auto Scaling groups associated with the node group.
+	AutoScalingGroups []AutoScalingGroup `json:"autoScalingGroup,omitempty"`
+
+	// The remote access security group associated with the node group. This security
+	// group controls SSH access to the worker nodes.
+	RemoteAccessSecurityGroup string `json:"remoteAccessSecurityGroup,omitempty"`
+}
+
+// AutoScalingGroup is an autoscaling group associated with a NodeGroup.
+type AutoScalingGroup struct {
+	// The name of the Auto Scaling group associated with an Amazon EKS managed
+	// node group.
+	Name string `json:"name,omitempty"`
+}
+
+// A NodeGroupSpec defines the desired state of an EKS NodeGroup.
+type NodeGroupSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  NodeGroupParameters `json:"forProvider"`
+}
+
+// A NodeGroupStatus represents the observed state of an EKS NodeGroup.
+type NodeGroupStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     NodeGroupObservation `json:"atProvider,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// A NodeGroup is a managed resource that represents an AWS Elastic Kubernetes
+// Service NodeGroup.
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="RECLAIM-POLICY",type="string",JSONPath=".spec.reclaimPolicy"
```

Comment:
Similar to `NodePool` in GKE, it could be useful to show the cluster name as printer column.

Reply:
SGTM 👍